### PR TITLE
Makefile to build targets

### PR DESCRIPTION
**Pull request by sylv-io**
Implement GNU make environment to build targets concurrently. The Makefile reflects  operations in `run.sh`.
Build artefacts can depend on specific configuration variable. By modifying the configuration file, only the related target are rebuild. (see definition CONFIG_DEP in Makefile)

Tested on:
* Fedora 33 (gcc 10.2.1)
* Ubuntu 20.10 (gcc 8.4.0)

### CODE REVIEW

**Comment by sylv-io** [comment]
Still WIP

**Comment by sylv-io** [comment]
Now it is ready for review :+1:

**Comment by sylv-io** [comment]
This PR depends on #64. I will combine and squash some commits.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+	@echo  '  debian             - Build reproducible Debian Buster'
+	@echo  '  ubuntu/ubuntu-18   - Build reproducible Ubuntu Bionic (latest)'
+	@echo  '  ubuntu-20          - Build reproducible Ubuntu Focal'
+	@echo  '  sign               - Sign OS packages'
+	@echo  '  upload             - Upload OS package to provisioning server'
+	@echo  '*** Run in QEMU'
+	@echo  '  mbr-run            - Run MBR bootloader'
+	@echo  '  efi-run            - Run EFI application'
+	@echo  '*** MISC'
+	@echo  '  default            - Generate default run.config'
```
Comment (jensdrenhaus) [inline Makefile:117]: default-config

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+	@echo  '  efi-image          - Build EFI application image'
+	@echo  '*** Install toolchain'
+	@echo  '  go-tools           - Build/Update Golang tools'
+	@echo  '  tboot              - Build tboot'
+	@echo  '  debos              - Create all docker debos environments'
+	@echo  '  debos-debian       - Create docker debos environment for debian'
+	@echo  '  debos-ubuntu	     - Create docker debos environment for ubuntu'
+	@echo  '*** Build Operating Sytem'
+	@echo  '  debian             - Build reproducible Debian Buster'
+	@echo  '  ubuntu/ubuntu-18   - Build reproducible Ubuntu Bionic (latest)'
```
Comment (jensdrenhaus) [inline Makefile:109]: ubuntu-18

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+	@echo
+	@echo  '*** system-transparency targets ***'
+	@echo  '  Use "make [target] V=1" for extra build debug information'
+	@echo  '*** Build image'
+	@echo  '  all                - Build all image formats'
+	@echo  '  mbr-image          - Build MBR boatloader image'
+	@echo  '  efi-image          - Build EFI application image'
+	@echo  '*** Install toolchain'
+	@echo  '  go-tools           - Build/Update Golang tools'
+	@echo  '  tboot              - Build tboot'
```
Comment (jensdrenhaus) [inline Makefile:103]: actually tboot does not belog to toolchain but to Operating System

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+help:
+	@echo
+	@echo  '*** system-transparency targets ***'
+	@echo  '  Use "make [target] V=1" for extra build debug information'
+	@echo  '*** Build image'
+	@echo  '  all                - Build all image formats'
+	@echo  '  mbr-image          - Build MBR boatloader image'
+	@echo  '  efi-image          - Build EFI application image'
+	@echo  '*** Install toolchain'
```
Comment (jensdrenhaus) [inline Makefile:101]: add target "toolchain"

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+include $(top)/stboot-installation/mbr-bootloader/makefile
+include $(top)/stboot-installation/efi-application/makefile
+
+help:
+	@echo
+	@echo  '*** system-transparency targets ***'
+	@echo  '  Use "make [target] V=1" for extra build debug information'
+	@echo  '*** Build image'
+	@echo  '  all                - Build all image formats'
+	@echo  '  mbr-image          - Build MBR boatloader image'
```
Comment (jensdrenhaus) [inline Makefile:99]: mbr-bootloader-installation

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+	@echo  '  efi-image          - Build EFI application image'
```
Comment (jensdrenhaus) [inline Makefile:100]: efi-application-installation

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+include $(top)/stboot-installation/common/makefile
+include $(top)/stboot-installation/mbr-bootloader/makefile
+include $(top)/stboot-installation/efi-application/makefile
+
+help:
+	@echo
+	@echo  '*** system-transparency targets ***'
+	@echo  '  Use "make [target] V=1" for extra build debug information'
+	@echo  '*** Build image'
+	@echo  '  all                - Build all image formats'
```
Comment (jensdrenhaus) [inline Makefile:98]: - Build all installation options

**Review by jensdrenhaus — CHANGES_REQUESTED** [review]
Add 
```
Build foo
Done foo
```
for all targets. (Maybe a line break, too)

File: Makefile (hunk 5 above)
Comment (sylv-io) [reply Makefile]: I also adapted the run target to "run-mbr-bootloader"

File: Makefile (hunk 6 above)
Comment (sylv-io) [reply Makefile]: I also adapted the run target to "run-efi-bootloader"

**Comment by sylv-io** [comment]
found some typos

**Comment by sylv-io** [comment]
now it is ready :smile:

**Review by jensdrenhaus — APPROVED** [review]
:rocket:

**Pull request merged**